### ISSUE TITLE
Details missing of an mp4 smartphone video

### ISSUE THREAD

**Issue by dml33** [photoprism/photoprism#1447]
Hi.

I am using PP for some months and there is something I cannot understand about how the mp4 videos of my Huawei P20 smartphone are imported in PP without any details. For example, the created date and the GPS location are missing.

However, exiftool shows both data.

For example, given an mp4 video recorded on 2021:06:12 17:30:39, when it is imported, its filename is changed to the imported time (20210614_175236) and their details are missing:

exiftool information:
```
root@Emby2:/var/db/photoprism/storage/originals/2021/06 # exiftool 20210614_175236_3B639289.mp4
ExifTool Version Number         : 12.00
File Name                       : 20210614_175236_3B639289.mp4
Directory                       : .
File Size                       : 306 MB
File Modification Date/Time     : 2021:06:14 17:52:37+00:00
File Access Date/Time           : 2021:06:14 17:52:36+00:00
File Inode Change Date/Time     : 2021:06:14 18:02:22+00:00
File Permissions                : rw-r--r--
File Type                       : MP4
File Type Extension             : mp4
MIME Type                       : video/mp4
Major Brand                     : MP4 v2 [ISO 14496-14]
Minor Version                   : 0.0.0
Compatible Brands               : isom, mp42
Movie Header Version            : 0
Create Date                     : 2021:06:12 17:30:39
Modify Date                     : 2021:06:12 17:30:39
Time Scale                      : 1000
Duration                        : 0:01:01
Preferred Rate                  : 1
Preferred Volume                : 100.00%
Preview Time                    : 0 s
Preview Duration                : 0 s
Poster Time                     : 0 s
Selection Time                  : 0 s
Selection Duration              : 0 s
Current Time                    : 0 s
Next Track ID                   : 3
GPS Coordinates                 : 40 deg 1' 25.32" N, 4 deg 4' 20.28" W
Android Version                 : 10
Track Header Version            : 0
Track Create Date               : 2021:06:12 17:30:39
Track Modify Date               : 2021:06:12 17:30:39
Track ID                        : 1
Track Duration                  : 0:01:01
Track Layer                     : 0
Track Volume                    : 0.00%
Image Width                     : 3840
Image Height                    : 2160
Graphics Mode                   : srcCopy
Op Color                        : 0 0 0
Compressor ID                   : avc1
Source Image Width              : 3840
Source Image Height             : 2160
X Resolution                    : 72
Y Resolution                    : 72
Bit Depth                       : 24
Pixel Aspect Ratio              : 65536:65536
Color Representation            : nclx 9 1 9
Video Frame Rate                : 30.01
Matrix Structure                : 1 0 0 0 1 0 0 0 1
Media Header Version            : 0
Media Create Date               : 2021:06:12 17:30:39
Media Modify Date               : 2021:06:12 17:30:39
Media Time Scale                : 48000
Media Duration                  : 0:01:01
Handler Type                    : Audio Track
Handler Description             : SoundHandle
Balance                         : 0
Audio Format                    : mp4a
Audio Channels                  : 2
Audio Bits Per Sample           : 16
Audio Sample Rate               : 48000
Media Data Size                 : 320050693
Media Data Offset               : 405182
Image Size                      : 3840x2160
Megapixels                      : 8.3
Avg Bitrate                     : 42 Mbps
GPS Latitude                    : 40 deg 1' 25.32" N
GPS Longitude                   : 4 deg 4' 20.28" W
Rotation                        : 90
GPS Position                    : 40 deg 1' 25.32" N, 4 deg 4' 20.28" W
```

Details:

![image](https://user-images.githubusercontent.com/25033130/126912811-0d47be08-d7a4-42b7-b821-9b262ef27f2a.png)

Yaml sidecar file

```
TakenAt: 2021-06-14T17:52:36Z
UID: pqupe48qhwejfels
Type: video
Title: Portrait
OriginalName: 210612-VID_20210612_192936
TimeZone: UTC
Year: -1
Month: -1
Day: -1
Quality: 3
Details:
  Keywords: blue, portrait
CreatedAt: 2021-06-14T18:02:32.404019785Z
UpdatedAt: 2021-06-14T18:02:34.308152139Z
```

Why is this so?

Thanks!

**Comment by lastzero** [comment]
Sounds like the issue we fixed in #810... what PhotoPrism version are you using?

**Comment by dml33** [comment]
Hi.

I am using 210523-b1856b9d45502ba1a35e1d2ae6ca12fd17223895 version.

Just in case, the exiftool -j output:

```
root@Emby2:/var/db/photoprism/storage/originals/2021/06 # exiftool -j 20210614_175236_3B639289.mp4
[{
  "SourceFile": "20210614_175236_3B639289.mp4",
  "ExifToolVersion": 12.00,
  "FileName": "20210614_175236_3B639289.mp4",
  "Directory": ".",
  "FileSize": "306 MB",
  "FileModifyDate": "2021:06:14 17:52:37+00:00",
  "FileAccessDate": "2021:06:14 17:52:36+00:00",
  "FileInodeChangeDate": "2021:06:14 18:02:22+00:00",
  "FilePermissions": "rw-r--r--",
  "FileType": "MP4",
  "FileTypeExtension": "mp4",
  "MIMEType": "video/mp4",
  "MajorBrand": "MP4 v2 [ISO 14496-14]",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["isom","mp42"],
  "MovieHeaderVersion": 0,
  "CreateDate": "2021:06:12 17:30:39",
  "ModifyDate": "2021:06:12 17:30:39",
  "TimeScale": 1000,
  "Duration": "0:01:01",
  "PreferredRate": 1,
  "PreferredVolume": "100.00%",
  "PreviewTime": "0 s",
  "PreviewDuration": "0 s",
  "PosterTime": "0 s",
  "SelectionTime": "0 s",
  "SelectionDuration": "0 s",
  "CurrentTime": "0 s",
  "NextTrackID": 3,
  "GPSCoordinates": "40 deg 1' 25.32\" N, 4 deg 4' 20.28\" W",
  "AndroidVersion": 10,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2021:06:12 17:30:39",
  "TrackModifyDate": "2021:06:12 17:30:39",
  "TrackID": 1,
  "TrackDuration": "0:01:01",
  "TrackLayer": 0,
  "TrackVolume": "0.00%",
  "ImageWidth": 3840,
  "ImageHeight": 2160,
  "GraphicsMode": "srcCopy",
  "OpColor": "0 0 0",
  "CompressorID": "avc1",
  "SourceImageWidth": 3840,
  "SourceImageHeight": 2160,
  "XResolution": 72,
  "YResolution": 72,
  "BitDepth": 24,
  "PixelAspectRatio": "65536:65536",
  "ColorRepresentation": "nclx 9 1 9",
  "VideoFrameRate": 30.01,
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2021:06:12 17:30:39",
  "MediaModifyDate": "2021:06:12 17:30:39",
  "MediaTimeScale": 48000,
  "MediaDuration": "0:01:01",
  "HandlerType": "Audio Track",
  "HandlerDescription": "SoundHandle",
  "Balance": 0,
  "AudioFormat": "mp4a",
  "AudioChannels": 2,
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 48000,
  "MediaDataSize": 320050693,
  "MediaDataOffset": 405182,
  "ImageSize": "3840x2160",
  "Megapixels": 8.3,
  "AvgBitrate": "42 Mbps",
  "GPSLatitude": "40 deg 1' 25.32\" N",
  "GPSLongitude": "4 deg 4' 20.28\" W",
  "Rotation": 90,
  "GPSPosition": "40 deg 1' 25.32\" N, 4 deg 4' 20.28\" W"
}]
```
Thanks

**Comment by lastzero** [comment]
I assume exiftool has NOT been disabled in your photoprism config? Does it work when indexing or is it just broken when importing?

You may want to try our development preview as it contains improvements for video metadata, see release notes on docs.photoprism.org

**Comment by lastzero** [comment]
Any warnings or errors in your logs? See https://demo.photoprism.org/library/errors for example.

**Comment by dml33** [comment]
> I assume exiftool has NOT been disabled in your photoprism config? 
How can I see it?

I only have this issue with mp4 videos. However, it works perfectly with photos (taken with the same smartphone), where all EXIF data is extracted and correctly placed in the details section.

> Does it work when indexing or is it just broken when importing?

With this PP version, I have only imported photos and videos. 

But, previously, with the 210505-d3e53a89 version, I imported and even re-indexed them a couple of times. It was the same:  mp4 video details were always empty, but they were correct for the photos.

**Comment by lastzero** [comment]
> I only have this issue with mp4 videos. However, it works perfectly with photos (taken with the same smartphone), where all EXIF data is extracted and correctly placed in the details section.

PhotoPrism natively supports Exif, but not video metadata which is not Exif. It needs to run Exiftool for this - other than the name suggests, it also supports other metadata formats as found in videos and XMP.

> > Does it work when indexing or is it just broken when importing?
> 
> With this PP version, I have only imported photos and videos.
> 
> But, previously, with the 210505-d3e53a89 version, I imported and even re-indexed them a couple of times. It was the same: mp4 video details were always empty, but they were correct for the photos.

Would be great if you can try this on a test installation with a few files, ideally also with our development preview. You may also send example files for testing to hello@photoprism.app. Note we're out of office this week, so it may take a while before we can debug this.

**Comment by lastzero** [comment]
It may also be a permission issue with your cache or sidecar folder, e.g. if it isn't writable by the user PhotoPrism runs as. Thus you should check for [warnings or errors in your logs](https://demo.photoprism.org/library/errors) in any case.

**Comment by lastzero** [comment]
You can see your config options in Settings > Advanced:
![Screenshot_20210726_111509](https://user-images.githubusercontent.com/301686/126964962-53768e5e-2403-455d-96a0-0e56e92a9228.jpg)

**Comment by dml33** [comment]
Hi.

Exiftool has been always enabled in Advanced settings:

![image](https://user-images.githubusercontent.com/25033130/127035852-cbcfb045-21eb-4881-96de-7b60c537d1c7.png)

As you suggested, I re-indexed the whole month and now videos details are properly imported! 👯 

![image](https://user-images.githubusercontent.com/25033130/127037677-141afef6-cf52-4b45-9aca-f5ac1a536906.png)

Does PP use its own exiftool tool? Or does it need the exiftool command for videos? I say this, because yesterday I installed the exiftool package in my FreeBSD system (which was not installed) and then I extracted the json reports I attached.

Anyway, later I will import new videos and see what happens. Maybe i was the re-index process or the exiftool package... I do not know.

Thanks

**Comment by lastzero** [comment]
You need to have exiftool installed. I was assuming you run our Docker image which comes with exiftool pre-installed.

**Comment by dml33** [comment]
I am using the [FreeBSD port](https://github.com/huo-ju/photoprism-freebsd-port) to use it within a jail in a XigmaNAS system.

Well, I will speak with the owner, since exiftool package is not set as a dependency and it is not installed.

Why it is not needed for photos, since their EXIF data was extracted correctly without exiftool?

I deleted the videos and re-imported, trying to avoid the video filenames to be changed with the timestamp when they were firstly imported. Well, the video details are taken correctly 👍 , but they were renamed to the wrong ones 🙅 

```
2021-07-26 20:51:07 INFO import: updated related jpg file /var/db/photoprism/storage/sidecar/2021/06/20210614_173723_E70723B0.mp4.jpg

2021-07-26 20:51:05 INFO import: updated main mp4 file 2021/06/20210614_173723_E70723B0.mp4

2021-07-26 20:51:01 INFO converting 2021/06/20210614_173723_E70723B0.mp4 to jpg

2021-07-26 20:51:01 INFO import: moving main mp4 file 210612-VID_20210612_181833.mp4 to 2021/06/20210614_173723_E70723B0.mp4

2021-07-26 20:51:01 INFO media: 210612-VID_20210612_181833.mp4 was taken at 2021-06-12 14:18:44 +0000 UTC (meta)
```

Can I fix this?

Thanks!

**Comment by lastzero** [comment]
PhotoPrism supports Exif natively, without Exiftool. Videos however don't use Exif but other standards. There might still be old metadata in your index or sidecar files, even though you deleted the videos. Try with a clean test Installation to see if the problem persists.

**Comment by dml33** [comment]
Thanks for your support!

If I wanted to take the whole information (new labels and adjustments) about the photos and videos I already have, how would I make a new installation pick it? Copying the originals and sidecar folders and re-import/re-index?

Thanks

**Comment by dml33** [comment]
Hi.

I still have some issues. The exiftool reports the creation time in UTC:

```
  "MediaCreateDate": "2021:06:12 17:30:39",
  "MediaModifyDate": "2021:06:12 17:30:39",
```

Since I am in Spain, our local time is GMT+2.

But, the local time which appears in the UI details is actually that UTC time, but not the right local time:

![image](https://user-images.githubusercontent.com/25033130/127195516-b57f0856-3808-4705-a800-814f15d65e13.png)

The correct local time shall be 19:30:30 and the UTC time shall be 17:30:39.

Another fact is that the video filename, always in local time is 210612-VID_20210612_192936.mp4

In case of photos, the local time and UTC time are correctly calculated.

Thanks

**Comment by lastzero** [comment]
See Exiftool: Set QuickTimeUTC flag for fixing time zone of videos #1388

**Comment by dml33** [comment]
Great! When is next release of PP planned to be released?

**Comment by graciousgrey** [comment]
As soon as we finished facial recognition :)
